widget/embedded/src/store/utils/data.test.ts

```diff
@@ -0,0 +1,101 @@
+import type { Token } from 'rango-sdk';
+
+import { describe, expect, test } from 'vitest';
+
+import { createToken } from '../../test-utils/fixtures';
+import { createTokenHash } from '../../utils/meta';
+
+import { calculateSupportedTokens } from './data'; // Adjust path as necessary
+
+const tokens: Token[] = [
+  { ...createToken(), blockchain: 'blockchainA' },
+  { ...createToken(), blockchain: 'blockchainA' },
+  { ...createToken(), blockchain: 'blockchainB' },
+  { ...createToken(), blockchain: 'blockchainC' },
+];
+
+const mockData: Parameters<typeof calculateSupportedTokens>[0] = {
+  type: 'source',
+  meta: {
+    tokensMapByTokenHash: new Map(),
+    tokensMapByBlockchainName: {},
+  },
+  config: { tokens: [], blockchains: undefined },
+};
+
+tokens.forEach((token) => {
```

Comment:
you can make this part of code a function called `generateTokensMapByBlockchainName` or whatever.
signature:
```
generateTokensMapByBlockchainName(tokens: Token[]): Record<Blockchain, TokenHash[]>
```
It will be a pure function and you can run it to generate hashes in your tests:

```
test( ... , () => {
mockData.meta.tokensMapByBlockchainName = generateTokensMapByBlockchainName(...)
})
```

Reply:
I also really don't think making a global object and mutating it in every test is a good idea. Making the tests fragile and hard to read.

My suggestion is moving `mockData` to each tests and only keep `const tokens` globally. You should only make a util for `mockdata.meta` to be reuse in all your tests. In my previous comment I described what it can look like.